src/main/java/edu/bu/metcs673/trackr/service/UserService.java

```diff
@@ -17,10 +19,19 @@
 @Component
 public interface UserService {
 
-	public List<User> retrieveAllUsers();
-
+	// TODO: REMOVE this method, it is used for testing purposes only (HelloWorldController)
 	public User findUserById(long id);
 
-	public void createUser(String username, String password, String email);
+	/**
+	 * Creates a new record in the USERS table with the provided method parameters.
+	 * These parameters are validated and if the validations fail, then no record is
+	 * saved and an error is returned.
+	 * 
+	 * @param username
+	 * @param password
+	 * @param email
+	 * @return 
+	 */
+	public ResponseEntity<GenericApiResponse> createUser(User userInput);
```

Comment:
Instead of using `ResponseEntity` in the service model should we instead only user User (domain models)? My thinking is that the controller needs to know about the API response entity and the UserService only operates on domain objects.

Reply:
Ok, I could modify the method to return the Object and then modify the controllers to use the @ResponseBody annotation